docs/examples/UserGuide/create_from_func.jl

```diff
@@ -55,4 +55,45 @@ gen_cube = mapCube(g, (lon, lat, time);
 # !!! info "slicing dim"
 #     Note that now the broadcasted dimension is `lon`.
 
-gen_cube.data[:,:,1]
\ No newline at end of file
+gen_cube.data[:, :, 1]
+
+# ## Generate Cube: Modify dimensions
+
+# One can also change the dimensions of a YAXArray.
+# For example, the location of `gen_cube` can be also expressed as a region or area of interest.
+# The following example transforms the raster data cube with two spatial dimensions,
+# i.e., longitude and latitude, into a [vector data cube](https://r-spatial.org/r/2022/09/12/vdc.html) with just one spatial dimension i.e. the region (e.g. district or country).
+
+# First, create a function `geo_region` to calculate the region for given geographical coordinates. In practice, one might use a spatial join of points to polygons here.
+
+function get_region(lon, lat)
+    1 <= lon < 10 && 1 <= lat < 5 && return "A"
+    1 <= lon < 10 && 5 <= lat < 10 && return "B"
+    10 <= lon < 15 && 1 <= lat < 5 && return "C"
+    return "D"
+end
+
+# Next, create a function `transform_cube` that will aggregate the values for any given time step.
+# `xin` is the raster data at one time step that is a Matrix with values `xin[lon, lat]`
+# `xout` is the vector data at one time step that is a Vector with values `xout[region]`
+
+function transform_cube(xout, xin, regions, points_to_regions_matrix)
+    result = Vector{eltype(xin)}(undef, length(xout))
+    for (i, region) in enumerate(regions)
+        region_points = findall(isequal(region), points_to_regions_matrix)
```

Comment:
This step here is allocating and not depending on the time step. I think even for documentation purposes it would be good to move this preprocessing step out of the loop. You could define 

````julia
region_to_indexlist = map(enumerate(regions)) do (i,region)
    findall(isequal(region), points_to_regions_matrix)
end
````

as preprocessing outside the loop and passing the result to mapCube and then inside `transform_cube` you just do
````julia
for i in 1:length(regions)
    xout[i] = sum(view(xin,region_to_indexlist[i]))
end
````
Should avoid all allocations inside the loop